### PR TITLE
Allow a custom set of root certificates for verifying server certificates

### PR DESCRIPTION
**Pull request by nwoodmsft**
Hello! I would like to contribute a bug fix and a new feature to the project:

### Fix: The _--no-check-cert_ option (InsecureSkipVerify) is not respected by cmd.follow()

  * I noticed was that even if the _--no-check-cert_ option was requested, the cmd.follow() function did not respect it and would still fail if the destination server certificate is untrusted. 
  * Example of the issue:

```
$ ./getparty --no-check-cert https://insecure.server.com
2021/01/24 03:19:08 GET: https://insecure.server.com
unexpected exit: run: follow: Get "https://insecure.server.com": x509: certificate signed by unknown authority
```

### Feature: Support for providing a custom set of root certificates to verify server certificates. 

  * This PR introduces a new option called _--certs-file_ which can be used to specify a custom certificate file or certificate bundle file containing root certificates. These certificates will be set as the http.Transport.TLSClientConfig.RootCAs so that the http client will trust them and use them to verify the server. 
  * This enables the user to connect to servers with self-signed certificates without having to resort to using --no-check-cert
  * Example of the new feature/option:

```
$ ./getparty https://insecure.server.com
2021/01/24 03:19:08 GET: https://insecure.server.com
unexpected exit: run: follow: Get "https://insecure.server.com": x509: certificate signed by unknown authority

$ ./getparty --certs-file=customcerts.crt https://insecure.server.com
insecure.server.com
2021/01/24 03:30:13 GET: https://insecure.server.com
2021/01/24 03:30:13 HTTP response: 200 OK
Length: 819 (819.0b) [text/html]
Saving to: "insecure.server.com"

P01 409.0b     100% ============================================================== Avg: 513.2KiB/s Peak: 2.6MiB/s
P02 410.0b     100% ============================================================== Avg: 473.0KiB/s Peak: 7.5MiB/s
Concatenating: 100% ============================================================== Done

2021/01/24 03:30:13 "insecure.server.com" saved [819/819]
```

Thank you!

### CODE REVIEW

File: getparty.go [hunk 1]
```diff
@@ -424,6 +433,24 @@ func (cmd Cmd) applyHeaders(req *http.Request) {
 	}
 }
 
+func (cmd Cmd) getTransport() (*http.Transport, error) {
+	transport := cleanhttp.DefaultPooledTransport()
```
Comment (vbauerster) [inline getparty.go:437]: `cleanhttp` provides two options `DefaultTransport` and`DefaultPooledTransport`.
Can you please add `pooled bool` param and based on it make either one.
Follow method used `cleanhttp.DefaultClient` which makes `DefaultTransport` under the hood.

File: getparty.go [hunk 2]
```diff
@@ -332,8 +333,16 @@ func (cmd Cmd) follow(jar http.CookieJar, userUrl string) (session *Session, err
 			jar.SetCookies(u, cookies)
 		}
 	}
-	client := cleanhttp.DefaultClient()
-	client.Jar = jar
+
+	transport, err := cmd.getTransport()
```
Comment (vbauerster) [inline getparty.go:337]: follow method should pass `pooled = false` to `cmd.getTransport()`, see my comment below.

File: getparty.go [hunk 3]
```diff
@@ -252,10 +254,9 @@ func (cmd *Cmd) Run(args []string, version string) (err error) {
 	)
 
 	var eg errgroup.Group
-	transport := cleanhttp.DefaultPooledTransport()
-	transport.TLSHandshakeTimeout = time.Duration(cmd.options.Timeout) * time.Second
-	if cmd.options.InsecureSkipVerify {
-		transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	transport, err := cmd.getTransport()
```
Comment (vbauerster) [inline getparty.go:257]: here we should get `pooled = true` transport, see my comment on `getTransport` method below.

File: getparty.go [hunk 4]
```diff
@@ -332,8 +333,16 @@ func (cmd Cmd) follow(jar http.CookieJar, userUrl string) (session *Session, err
...
+	if err != nil {
+		return nil, err
+	}
+	client := &http.Client{
+		Transport: transport,
+		Jar:       jar,
+	}
+
 	client.CheckRedirect = func(*http.Request, []*http.Request) error {
```
Comment (vbauerster) [inline getparty.go:346]: this one could be spliced to above client instantiation.

**Review by vbauerster — CHANGES_REQUESTED** [review]
Thanks for contribution! This is good catch.
Can you please check my comments? Apart from that everything else is ok.

File: getparty.go (hunk 4 above)
Comment (nwoodmsft) [reply getparty.go]: I think I understood - please do double check my commit though :)

**Comment by nwoodmsft** [comment]
Thank you @vbauerster - good catch. I made your suggested edits and pushed a new commit.

**Review by vbauerster — APPROVED** [review]

**Pull request merged**